kroxylicious/src/main/java/io/kroxylicious/proxy/internal/DefaultFilterContext.java

```diff
@@ -209,6 +217,14 @@ public void forwardResponse(ApiMessage response) {
         }
     }
 
+    @Override
+    public void abortConnection(Throwable t) {
```

Comment:
I think the usage of this makes sense, but I'm not sure we want to expose `abortConnection` to all filter authors as part of the public API.

I think there is a wider conversation to be had around how we handle failure in filters and how filters may want to signal that a connection is in a bad state. 

Could we do something hacky and tactical for now such as casting to `DefaultFilterContext` at the call site for access to abort connection? Or are we happy to add this to the public API and remove later if we decide on an exception based flow?

Reply:
After discussion, we switched to synchronous reconciliation to side step #351 so the immediate need for `abortConnection` goes.